### PR TITLE
Fix bugs encountered (mostly in battles) in Ara Fell

### PR DESCRIPTION
**Pull request by Ghabry**
This fixes lots of bugs I found while playing through Ara Fell and defeating the first boss. The biggest change was moving most Interpreter commands to the Interpreter (from map) because this breaks battle otherwise.
Many visual Problems were fixed in battle.
If I didn't fail in cut-paste the Interpreter should work as good as before.

The map is here but for some reason when starting the battle it crashes (works locally)
https://easy-rpg.org/play/?game=af_demo&start-map-id=223&start-party=1&test-play

I also fixed a buffer Overflow in vsnprintf which occured in Spanish Dragonball Z because the map tree was too deep.
And double clicking in the Finder on MacOS should work now...

Not ready yet. I need somebody to test it on a Mac ;)

### CODE REVIEW

File: src/game_interpreter.cpp [hunk 1]
```diff
@@ -907,6 +908,46 @@ bool Game_Interpreter::CommandControlVariables(RPG::EventCommand const& com) { /
...
+				break;
+			case 6:
+				// Enemy Spirit
+				value = (*Main_Data::game_enemyparty)[com.parameters[5]].GetSpi();
+				break;
+			case 7:
+				// Enemy Agility
+				value = (*Main_Data::game_enemyparty)[com.parameters[5]].GetAgi();
+				break;
+			}
```
Comment (Ghabry) [inline src/game_interpreter.cpp]: Look what we forgot :o

**Comment by Ghabry** [comment]
Regression in EHR is fixed.

The only problem I'm aware of in Ara Fell is that the "Scout" battle command is behaving totally wrong :/

File: src/graphics.cpp [hunk 2]
```diff
@@ -227,6 +227,11 @@ void Graphics::Freeze() {
 }
 
 void Graphics::Transition(TransitionType type, int duration, bool erase) {
+	if (erase == screen_erased) {
+		// Don't allow another erase or non-erase
+		return;
+	}
+
```
Comment (Zegeri) [inline src/graphics.cpp]: This doesn't reproduce the behaviour of RPG_RT when using Show Screen in a non-erased screen. For example, try using Show Screen with the transition type Scroll Right.

**Comment by Ghabry** [comment]
Enough fixes for this PR. The next commits will fix regressions.

The first boss can be defeated now. That boss absorbs other enemies to generate a "shield". The Interpreter checked if all enemy HP was 0, then the real boss can be attacked. Due to different bugs the sum was never 0.

ForceFlee is not needed, see it as a bonus ;)

**Comment by fdelapena** [comment]
First regression found: battler sprites are the same for all allied in EHR:
The allied character is the same for all

Force flee will unbreak multiple games, nice bonus :+1:.

**Comment by Ghabry** [comment]
RPG2k3 supports many ways to do animations.

1. The "right hand" animation fades out the battler and then it stays at an empty frame (old behaviour: it was looping or stuck at the last non-empty frame)
2. The battle animation plays
3. Not implemented: A "fade-out" animation is played (is an extra per-actor setting), not used by many games but Ara Fell does. This makes Lita's attacks look a bit strange.
4. The idle (or state) animation plays for the battler

**Comment by Ghabry** [comment]
Got confirmation that #705 is fixed

**Comment by Ghabry** [comment]
Fixed the reported regressions and solved more (easy to resolve) issues I encoutered.

**Comment by Ghabry** [comment]
\*pfew\* this is ready now :+1:

Battle should be a bit better now, also in other games ;)

**Pull request merged**